### DISCUSSION TITLE
PR2040 Spindle Sync?

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/RP2040#4] [General]
In looking at the Pico, would be possible to do spindle sync?  The design I have for a breakout board has I2C (GP19-21) and SPI (GP10-13) headers.  Each has at least 3 contiguous GPIO pins.  It is possible (via PIO?) to do spindle sync with either of those pin sets.  Of course, you lose the underlying I2C or SPI bus. For I2C, you also loose EEPROM. But I could imagine a scenario where users might make that trade-off.  I could imagine a small daughter card that has connectors and does any necessary signal conditioning.

**Comment by terjeio** [comment]
It would but likely not with my current code as it uses floating point math in an interrupt context. But then there is a second core doing nothing...

Signal conditioning is not needed since inputs can be configured for schmitt trigger mode ? I have photointerrupters for the encoder, these can be connected directly? Same goes for Hall sensors?

Note that I am not too keen to add the more advanced stuff such as spindle sync support to every driver, code maintenance and support needs to be factored in. Some of the advanced functionality should be reserved for higher end processors and breakout boards?

**Comment by undingen** [comment]
Hi, 
I saw that there are now commits which look like there is now spindle sync support for the rp2350 https://github.com/grblHAL/RP2040/commit/5b0938954b4160beee754c2573e8cad5ba64a550. Is this feature finished? Thinking about maybe using this board https://github.com/phil-barrett/RP23CNC/discussions/11 to convert my mini lathe and checking which options I have.

**Comment by terjeio** [comment]
@undingen I should be finished, but be aware that I have only tested the code with my simulator - not in a lathe.
